### PR TITLE
Fixed bugs with query merging

### PR DESCRIPTION
**Pull request by Poincare**
From @amandajliu's work w/ GitHunt, we discovered that there were some issues with the way query merging handled arrays and null values in the response. This PR fixes those issues.

TODO:

- [x] Update CHANGELOG.md with your change
- [x] Make sure all of the significant new logic is covered by tests
- [x] Rebase your changes on master so that they can be merged easily
- [x] Make sure all tests and linter rules pass

### CODE REVIEW

File: src/batching/queryMerging.ts [hunk 1]
```diff
@@ -157,25 +162,53 @@ export function unpackDataForRequest({
...
+        if (isNull(childData)) {
+          const selectionRet = unpackDataForRequest(assign(fieldOpts, {
+            startIndex: currIndex,
+          }) as UnpackOptions);
+          currIndex = selectionRet.newIndex;
+        } else if (isArray(childData)) {
+          const resUnpacked = [];
+          let newIndex = 0;
+
+          data[stringKey].forEach((dataObject) => {
```
Comment (stubailo) [inline src/batching/queryMerging.ts]: What's the difference between `childData` and `data[stringKey]`?

File: src/batching/queryMerging.ts [hunk 2]
```diff
@@ -157,25 +162,53 @@ export function unpackDataForRequest({
       const realName = resultKeyNameFromField(field);
       const aliasName = getOperationDefinitionName(getQueryDefinition(request.query), queryIndex);
       const stringKey = topLevel ? `${aliasName}___fieldIndex_${currIndex}` : realName;
-      unpackedData[realName] = data[stringKey];
       if (topLevel) {
         currIndex += 1;
       }
 
+      let childData = isNull(data) ? null : data[stringKey];
```
Comment (stubailo) [inline src/batching/queryMerging.ts]: Make this const, and create a new variable for the data that is going to be put into the unpacked response.

**Comment by Poincare** [comment]
Tested with GitHunt. Should be ready to merge.

**Comment by Poincare** [comment]
Tested with GitHunt. Works + should be ready to merge, @stubailo

**Pull request merged**